synth-methods/PATE-GAN/pate_gan_main.py

```diff
@@ -0,0 +1,203 @@
+        """
+
+        # Read data and metadata
+        if verbose:
+            print("\n[INFO] Reading input data and metadata from disk")
+        input_data, metadata = self.read_data(csv_path, metadata_json_path, verbose)
+        print(f"[INFO] #rows before removing NaN: {len(input_data)}")
+        input_data.dropna(inplace=True)
+        print(f"[INFO] #rows after removing NaN: {len(input_data)}")
+        with open(metadata_json_path) as metadata_json:
+            self.metadata = json.load(metadata_json)
+
+        if verbose:
+            print("\nSUMMARY")
+            print("-------")
+            print("Dataframe dimensions\n#rows:    {}\n#columns: {}".format(input_data.shape[0],
+                                                                            input_data.shape[1]))
+            print("\nColumn name \t Type\n" + 11 * "--")
+            for col in metadata['columns']:
+                print("{} \t {}".format(col['name'], col['type']))
+
+        # Extract parameters from json parameter file
+        with open(parameters_json_path) as parameters_json:
+            self.parameters = json.load(parameters_json)
+        self.num_samples_to_fit = self.parameters['parameters']['num_samples_to_fit']
+        self.random_state = self.parameters['parameters']['random_state']
+        self.num_datasets_to_synthesize = self.parameters["parameters"]["num_datasets_to_synthesize"]
+        self.num_samples_to_synthesize = self.parameters["parameters"]["num_samples_to_synthesize"]
+        self.batch_size = self.parameters["parameters"]["batch_size"]
+        self.num_teachers = self.parameters["parameters"]["num_teachers"]
+        self.num_iters = self.parameters["parameters"]["num_iters"]
+        self.learning_rate = self.parameters["parameters"]["learning_rate"]
+        self.eps = self.parameters['parameters']['epsilon']
+        self.delta = self.parameters['parameters']['delta']
+        
+        if "columns_to_synthesize" in self.parameters["parameters"].keys():
+            self.columns_to_synthesize = self.parameters["parameters"]["columns_to_synthesize"]
+        else:
+            self.columns_to_synthesize = "all"
+        
+        if not self.columns_to_synthesize in [None, "all", "None"]:
+            input_data = input_data[self.columns_to_synthesize]
+            metadata_update = {"columns": []}
+            self_metadata_update = {"columns": []}
+            for one_col in metadata["columns"]:
+                if one_col["name"] in self.columns_to_synthesize:
+                    metadata_update["columns"].append(one_col)
+                    self_metadata_update["columns"].append(one_col)
+            metadata = metadata_update
+            self.metadata = self_metadata_update
+
+        if verbose:
+            print(f"\n[INFO] Reading PATE-GAN parameters from json file:\n"
+                  f"num_samples_to_fit = {self.num_samples_to_fit}\n"
+                  f"random_state = {self.random_state}\n")
```

Comment:
Should the other main PATE-GAN parameters be printed here?

Reply:
Done